rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -805,6 +809,260 @@ private void assertEventWasEmittedWithExpectedData(byte[] expectedData) {
+            saveSvpFundTransactionUnsigned(svpFundTransaction);
+
+            return svpFundTransaction;
+        }
+
+        private BtcTransaction createSvpFundTransaction() {
+            BtcTransaction svpFundTransaction = new BtcTransaction(btcMainnetParams);
+
+            Sha256Hash parentTxHash = BitcoinTestUtils.createHash(1);
+            addInput(svpFundTransaction, parentTxHash);
+
+            svpFundTransaction.addOutput(spendableValueFromProposedFederation, proposedFederation.getAddress());
+            Address flyoverProposedFederationAddress =
+                PegUtils.getFlyoverAddress(btcMainnetParams, bridgeMainNetConstants.getProposedFederationFlyoverPrefix(), proposedFederation.getRedeemScript());
+            svpFundTransaction.addOutput(spendableValueFromProposedFederation, flyoverProposedFederationAddress);
+
+            return svpFundTransaction;
+        }
+
+        private void saveSvpFundTransactionUnsigned(BtcTransaction svpFundTransaction) {
+            savePegoutIndex(svpFundTransaction);
+            saveSvpFundTransactionHashUnsigned(svpFundTransaction.getHash());
+        }
+
+        private BtcTransaction createPegout() {
+            BtcTransaction pegout = new BtcTransaction(btcMainnetParams);
+            Sha256Hash parentTxHash = BitcoinTestUtils.createHash(2);
+            addInput(pegout, parentTxHash);
+            addOutputChange(pegout);
+
+            return pegout;
+        }
+
+        private void addInput(BtcTransaction transaction, Sha256Hash parentTxHash) {
+            // we need to add an input that we can actually sign,
+            // and we know the private keys for the following scriptSig
+            Federation federation = P2shErpFederationBuilder.builder().build();
+            Script scriptSig = federation.getP2SHScript().createEmptyInputScript(null, federation.getRedeemScript());
+
+            transaction.addInput(parentTxHash, 0, scriptSig);
+        }
+
+        private void signInputs(BtcTransaction transaction) {
+            List<BtcECKey> keysToSign =
+                BitcoinTestUtils.getBtcEcKeysFromSeeds(new String[]{"member01", "member02", "member03", "member04", "member05"}, true);
+            List<TransactionInput> inputs = transaction.getInputs();
+            for (TransactionInput input : inputs) {
+                BitcoinTestUtils.signTransactionInputFromP2shMultiSig(transaction, inputs.indexOf(input), keysToSign);
+            }
+        }
+
+        private void addOutputChange(BtcTransaction transaction) {
+            // add output to the active fed
+            Script activeFederationP2SHScript = activeFederation.getP2SHScript();
+            transaction.addOutput(Coin.COIN.multiply(10), activeFederationP2SHScript);
+        }
+
+        private void savePegoutIndex(BtcTransaction pegout) {
+            // save sigHash in pegout index
+            Optional<Sha256Hash> inputSigHash = BitcoinUtils.getFirstInputSigHash(pegout);
```

Comment:
There is a warning because the optional is being used without checking whether it is present.

Reply:
updated! Thanks